modules/core/src/smithy4s/Blob.scala

```diff
@@ -41,78 +117,171 @@ object Blob {
     string.getBytes(StandardCharsets.UTF_8)
   )
 
-  trait Encoder[A] {
-    def encode(a: A): Blob
-  }
+  final class ByteArrayBlob(val arr: Array[Byte]) extends Blob {
```

Comment:
This one is a special case of `ArraySliceBlob` right? Is it worth to make it a separate case in the ADT?

Reply:
Good point